cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1002,7 +1018,26 @@ private String getHiveDatabase(String namespace) {
 
     QueryHandle queryHandle = QueryHandle.generate();
     sessionConf.put(Constants.Explore.QUERY_ID, queryHandle.getHandle());
-    
+
+    Transaction tx = startTransaction();
+    ConfigurationUtil.set(sessionConf, Constants.Explore.TX_QUERY_KEY, TxnCodec.INSTANCE, tx);
+    ConfigurationUtil.set(sessionConf, Constants.Explore.CCONF_KEY, CConfCodec.INSTANCE, cConf);
+    ConfigurationUtil.set(sessionConf, Constants.Explore.HCONF_KEY, HConfCodec.INSTANCE, hConf);
+
+    return sessionConf;
+  }
+
+  protected Map<String, String> startSession(Id.Namespace namespace) throws IOException {
+    Map<String, String> sessionConf = Maps.newHashMap();
+
+    QueryHandle queryHandle = QueryHandle.generate();
+    sessionConf.put(Constants.Explore.QUERY_ID, queryHandle.getHandle());
+
+    String schedulerQueue = getNamespaceResolvedSchedulerQueue(namespace);
+    if (schedulerQueue != null) {
+      sessionConf.put(Constants.MapReduce.MAP_REDUCE_JOB_QUEUE_NAME, schedulerQueue);
```

Comment:
No need to add job queue name to session conf again, as `startSession` already does it, right?